app/src/main/java/com/dimagi/biometric/activities/EnrollActivity.java

```diff
@@ -1,33 +1,47 @@
 package com.dimagi.biometric.activities;
 
+import com.dimagi.biometric.OmniMatchUtil;
+
+import org.commcare.commcaresupportlibrary.BiometricUtils;
+import org.commcare.commcaresupportlibrary.identity.IdentityResponseBuilder;
+
 import java.util.ArrayList;
-import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
 
 import Tech5.OmniMatch.BioCommon;
+import Tech5.OmniMatch.FingerCommon;
 import Tech5.OmniMatch.MatcherCommon;
 
 public class EnrollActivity extends BaseActivity {
     @Override
     protected void onCaptureSuccess(MatcherCommon.Record activeRecord) {
-        // TODO: Need to add in position to returned data
-        List<byte[]> templateDataList = new ArrayList<>();
+        Map<BiometricUtils.BiometricIdentifier, byte[]> templateDataList = new HashMap<>();
         if (biometricType == BioCommon.BioType.Face) {
             byte[] templateData = activeRecord.getFace().getTemplateData().getData().toByteArray();
-            templateDataList.add(templateData);
+            templateDataList.put(BiometricUtils.BiometricIdentifier.FACE, templateData);
         } else {
             for (BioCommon.MatcherTemplate template : activeRecord.getNnFingersList()) {
                 byte[] templateData = template.getTemplateData().getData().toByteArray();
-                templateDataList.add(templateData);
+                templateDataList.put(
+                        OmniMatchUtil.getBiometricIdentifierFromPosition(FingerCommon.NistFingerPosition.values()[template.getPosition()]),
+                        templateData
+                );
             }
         }
-
-        // TODO: Return to CC with template(s). This is pending required changes to the CC Support Library
-        // IdentityResponseBuilder.registrationResponse(caseId).finalizeResponse(this);
+        String randId = UUID.randomUUID().toString();
+        IdentityResponseBuilder.registrationResponse(randId, templateDataList).finalizeResponse(this);
```

Comment:
@zandre-eng isn't the guid supposed to be the case ID?

Reply:
@avazirna Ah good catch. This is most likely still old code from testing. I have updated it to return the case ID instead ([105f4be](https://github.com/dimagi/biometric-android/pull/6/commits/105f4be791d93b6910d527e931b537c6829b6429))